indexer/db/db.go

```diff
@@ -209,26 +215,39 @@ func (d *Database) AddIndexedL1Block(block *IndexedL1Block) error {
 			return err
 		}
 
-		if len(block.Deposits) == 0 {
-			return nil
+		if len(block.Deposits) > 0 {
+			for _, deposit := range block.Deposits {
+				_, err = tx.Exec(
+					insertDepositStatement,
+					NewGUID(),
+					deposit.FromAddress.String(),
+					deposit.ToAddress.String(),
+					deposit.L1Token.String(),
+					deposit.L2Token.String(),
+					deposit.Amount.String(),
+					deposit.TxHash.String(),
+					deposit.LogIndex,
+					block.Hash.String(),
+					deposit.Data,
+				)
+				if err != nil {
```

Comment:
Just want to be sure that that an error wont result in really bad things, ie skipped indexing of deposits or withdrawals. Should we make a db transaction and then commit it at the end?

Reply:
It's safe against what you're describing :smile:.

The `txn` wrapper on line 206 wraps all this logic in a DB transaction. Any error in here will cause a rollback. Blocks are indexed one by one, so if the TX is rolled back the indexer will try to index the block again.